Upgrade to vagrant 1.8.4

ppp0 opened the pull request.

ppp0 commented:
retest this please

ppp0 commented:
@njam we were discussing that pinning to 1.7.3 is obsolete?

If yes, then let's move away from it if you agree (cc @kris-lab )

njam commented:
Yes, 1.8.1 should work.
But please verify that an SK build works successfully after the upgrade.

ppp0 commented:
* Updated to `1.8.3`, `rf -rf ~/.vagrant.d/`
* Problem installing plugins. Tested with:
  * `vagrant-phpstorm-tunnel` (some "resolver" problems at install, see below)
  * `vagrant-librarian-puppet` (ditto)

error dump:
```
vagrant plugin install vagrant-phpstorm-tunnel --plugin-version 0.1.10
Installing the 'vagrant-phpstorm-tunnel --version '0.1.10'' plugin. This can take a few minutes...
/opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:303:in `block in sort_dependencies': undefined method `payload' for nil:NilClass (NoMethodError)
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `each'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `sort_by'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:300:in `sort_dependencies'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:131:in `block (2 levels) in <class:Resolution>'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:431:in `push_state_for_requirements'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:423:in `require_nested_dependencies_for'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:411:in `activate_spec'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:353:in `attempt_to_swap_possibility'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:333:in `attempt_to_activate_existing_spec'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:318:in `attempt_to_activate'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:147:in `process_topmost_state'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolution.rb:72:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/vendor/molinillo/lib/molinillo/resolver.rb:42:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:201:in `start'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/resolver.rb:184:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:200:in `resolve'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:140:in `specs'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/definition.rb:129:in `resolve_remotely!'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:195:in `resolve_if_need'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:70:in `run'
	from /opt/vagrant/embedded/gems/gems/bundler-1.12.0/lib/bundler/installer.rb:22:in `install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:230:in `block in internal_install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:290:in `block in with_isolated_gem'
	from /opt/vagrant/embedded/lib/ruby/2.2.0/rubygems/user_interaction.rb:45:in `use_ui'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:289:in `with_isolated_gem'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:229:in `internal_install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/bundler.rb:102:in `install'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/plugin/manager.rb:62:in `block in install_plugin'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/plugin/manager.rb:72:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/plugin/manager.rb:72:in `install_plugin'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/action/install_gem.rb:37:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/warden.rb:34:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/builder.rb:116:in `call'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/runner.rb:66:in `block in run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/util/busy.rb:19:in `busy'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/action/runner.rb:66:in `run'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/base.rb:14:in `action'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/install.rb:32:in `block in execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/install.rb:31:in `each'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/install.rb:31:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/plugins/commands/plugin/command/root.rb:56:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/cli.rb:42:in `execute'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/lib/vagrant/environment.rb:302:in `cli'
	from /opt/vagrant/embedded/gems/gems/vagrant-1.8.3/bin/vagrant:174:in `<main>'
```

Reverted to `1.7.4` for the moment

njam commented:
On the mac?
I'm running 1.8.1 successfully since some time.
Just tried 1.8.4 and had the same problem!

But I rather meant to check that the SK builds on jenkins still works after we upgrade there.

ppp0 commented:
Checked and verified functioning on:
* Mac: Full removal (.vagrant.d/ dir) and reinstallation of plugins in the given versions
* Mandrill: ditto
* Some puppet-packages specs on mac
* Full sk specs on mandrill (to mimic jenkins)

ppp0 commented:
as in https://github.com/cargomedia/puppet-cargomedia/pull/1251#issuecomment-225858856

let's remove all but `boxes`

ppp0 commented:
@kris-lab please review

kris-lab commented on modules/vagrant/templates/install.sh:
```diff
@@ -1,5 +1,12 @@
 #!/bin/bash -e
 
 curl -sL "<%= @url %>" > vagrant.deb
+for i in $(find /usr /opt /home /var -ipath '*/.vagrant.d'); do
```
should we maybe do this only in `/home`?

ppp0 replied:
The thing is, on a Jenkins node, for instance, this dir is in `/var/lib/jenkins`

kris-lab replied:
ok, makes sense then!
